core/src/main/java/com/oath/oak/Slice.java

```diff
@@ -29,7 +29,8 @@
     // Allocation time version
     protected int version;
 
-    protected ByteBuffer buffer;
+    protected ByteBuffer readBuffer;
+    protected ByteBuffer writeBuffer;
```

Comment:
I currently do not see where writeBuffer and not readBuffer is used, and not both... What writeBuffer is used for?

Reply:
`writeBuffer` is used in `OakAttachedWriteBuffer`.
It must be stored in `Slice` because `OakAttachedWriteBuffer` is instantiated from an existing `Slice` object.

Reply:
Please add a comment about that, so it is clearer...

Reply:
We already have a comment for that in `Block`.
I'm against adding repeating comments in multiple places. If you feel this is necessary, then I suggest adding a comment in the form of "See Block for more details".